### DISCUSSION TITLE
converting Bruker/PrairieView multiple TIFF files + XML to NWB

### DISCUSSION THREAD

**Discussion by Thart92** [NeurodataWithoutBorders/helpdesk#47] [Q&A]
Hi, I'm trying to format my data for upload to NWB so that I can archive it on DANDI, but I'm finding myself a bit confused as to the best way to do this. DANDI recommends using NeuroConv to generate NWB files. However, I'm not very much of a python programmer, have not used NWB before, and find the documentation on NeuroConv unintuitive for my dataset (https://neuroconv.readthedocs.io/en/main/conversion_examples_gallery/imaging/tiff.html). For my own data processing and analysis I use ImageJ/Fiji.

My data are volumetric calcium imaging recordings (Bruker two-photon microscope) output from PrairieView software. Each imaging trial is stored as a folder containing 1320 2D TIFF files, corresponding to 33 z-planes and 40 frames per z-plane. There is also an XML file in the folder storing metadata about the trial overall and specifying the order/timing/z-position of each TIFF file within the overall trial.

Can someone please point me in the right direction for better understanding the conversion process for this sort of data, or otherwise getting started?

**Comment by bendichter** [comment]
Hi @Thart92, thanks for the issue! @weiglszonja just started working on a `BrukerTiffImagingInterface` [here](https://github.com/catalystneuro/neuroconv/pull/390) for a different lab. Could you coordinate with her? It would be great if this interface handled your Bruker data as welll.

To be honest, creating a new interface takes some time, so if you need this on a tighter schedule I would recommend working through [this tutorial](https://pynwb.readthedocs.io/en/stable/tutorials/domain/ophys.html#sphx-glr-tutorials-domain-ophys-py) on writing data in PyNWB. 

@weiglszonja, would you be able to coordinate with @Thart92, either to build the `BrukerTiffImagingInterface` to accomadate their needs or to help them through the 2p PyNWB tutorial?

**Reply by Thart92** [reply]
Thank you for the tutorial, that looks much more helpful than the example code I had found previously!

I'd be happy to work with @weiglszonja to help with the BrukerTiffImagingInterface, as well.

**Comment by bendichter** [comment]
The example code you pointed to is designed for datasets where all of the imaging data is in a single TIFF file, so I don't think it would be appropriate for your use-case.

**Reply by Thart92** [reply]
Yes, it seemed that way to me, but I wasn't sure if there was an obvious solution I was simply not seeing

**Comment by weiglszonja** [comment]
Hey @Thart92, I'd be happy to help you with your dataset. I'll have to adjust the `BrukerTiffImagingInterface` to handle volumetric data, and a good starting point for that would be for me to see some example data. If you could share the `XML` file (do you also have a `.env` file as well?) and some of the TIFF image files (could be the first one and a few others) then I could have a better sense how to modify the interface. Google Drive/Dropbox/Globus, anything like that works for data sharing. You can add me via my email, [szonja.weigl@catalystneuro.com](mailto:szonja.weigl@catalystneuro.com)
